tests/integration/attachment_test.go

```diff
@@ -59,15 +57,14 @@ func createAttachment(t *testing.T, session *TestSession, repoURL, filename stri
 func TestCreateAnonymousAttachment(t *testing.T) {
 	defer tests.PrepareTestEnv(t)()
 	session := emptyTestSession(t)
-	// this test is not right because it just doesn't pass the CSRF validation
-	createAttachment(t, session, "user2/repo1", "image.png", generateImg(), http.StatusBadRequest)
+	createAttachment(t, session, GetCSRF(t, session, "/user/login"), "user2/repo1", "image.png", generateImg(), http.StatusSeeOther)
```

Comment:
I don't know if someone looking at this code in two months knows why an attachment test calls a login endpoint...

Reply:
It reads as this: when `TestCreateAnonymousAttachment`, it needs to `GetCSRF` from `/user/login`, it sounds reasonable.

Reply:
If the CSRF token is not action related why can't we have a `GetCSRF` method without passed url? If `/user/login` is a redirect call `/user/settings`. One of both requests contains a token.

Reply:
It is user-related.

And I guess just no one has interest to work on https://github.com/go-gitea/gitea/pull/32130#issuecomment-2373616803 , and no one has the explicitly said that "that FIXME could be removed and Gitea will always use a common CSRF token for current user"

Reply:
I think what @KN4CK3R wants to say is `we should document exactly that with a comment`…

Reply:
I know it's user-related, that's the reason why there is the `session`.

```
func GetCSRF(t testing.TB, session *TestSession) string {
	t.Helper()
	
	req := NewRequest(t, "GET", "/user/login")
	resp := session.MakeRequest(t, req, NoExpectedStatus)
	
	if resp.Code == http.StatusSeeOther {
		req = NewRequest(t, "GET", "/user/settings")
		resp = session.MakeRequest(t, req, http.StatusOk)
	}
	
	doc := NewHTMLParser(t, resp.Body)
	return doc.GetCSRF()
}
```

If we decide to use action-related CSRF tokens the `GetCSRF` method with an url must change too because then a single page can contain multiple forms with different tokens but `doc.GetCSRF()` currently just returns the first token.

Reply:
It would slow down many tests by using the extra request.

In most cases, the test writer clearly knows what CSRF token they would like to use: anonymous, or a signed-in user.

If you prefer to do so, I think it needs 2 functions:

```
GetCSRFForAnonymous
GetCSRFForCurrentUser
```

Reply:
(while these improvements could be in a separate PR but not in this PR's scope)

Reply:
@KN4CK3R 

a separate PR
->  Refactor CSRF token #32216